packages/eslint-plugin-turbo/__tests__/cwd.test.ts

```diff
@@ -0,0 +1,78 @@
+import fs from "fs";
+import path from "path";
+import { execSync } from "child_process";
+
+describe("eslint settings check", () => {
+  beforeAll(() => {
+    const cwd = path.join(__dirname, "fixtures", "workspace");
+    execSync(`npm install`, { cwd });
```

Comment:
These tests are pretty heavy handed. This means we need to build first, and we're testing a lot more than just our config (all of eslint's caching essentially). 

Why not just validate the part we care about with something like:

```js
  it("sets correct env vars in settings", () => {
    // mock process.cwd()
    const spy = jest.spyOn(process, "cwd");
    spy.mockReturnValue(path.join(__dirname, "fixtures", "workspace"));
    // import the config 
    const config = require("../../lib/configs/recommended.ts");

    // validate it
    expect(config.settings).toEqual("...");
  });
```

Should be much faster / and way less brittle in the long run.

Reply:
I made an active decision to over-test because I'm worried about the flat config migration that ESLint is going through. I kinda wanted it to be brittle, and break if ESLint broke us. But your position is also reasonable: test the things we have control over.

My goal was to test that we successfully integrated into ESLint.

```Benchmark "test depends on build, acceptance test":
  Time (mean ± σ):      5.253 s ±  0.197 s    [User: 8.928 s, System: 1.349 s]
  Range (min … max):    5.051 s …  5.697 s    10 runs
 
Benchmark "test depends on build, no acceptance test":
  Time (mean ± σ):      3.093 s ±  0.130 s    [User: 4.415 s, System: 0.621 s]
  Range (min … max):    2.935 s …  3.370 s    10 runs
 
Benchmark "test doesn't depend on build, no acceptance test":
  Time (mean ± σ):      2.613 s ±  0.017 s    [User: 3.610 s, System: 0.457 s]
  Range (min … max):    2.593 s …  2.642 s    10 runs```

Timing:
- acceptance test: 2.16s
- build: 0.48s

Delta: 2.64s to add the acceptance test.

Shouldn't be flaky unless we aren't succeeding because of ESLint actually breaking us, which seems reasonable?

Reply:
That’s not as bad as I thought. Sounds good, I think this is fine. If they start being a pain to maintain we can rip it out and focus them a bit more in the future.